checks.d/vsphere.py

```diff
@@ -560,19 +560,26 @@ def _cache_morlist_raw_atomic(self, i_key, obj_type, obj, tags, regexes=None, in
         elif obj_type == 'datacenter':
             dc_tag = "vsphere_datacenter:%s" % obj.name
             tags_copy.append(dc_tag)
-            for compute_resource in obj.hostFolder.childEntity:
-                # Skip non-compute resource
-                if not hasattr(compute_resource, 'host'):
-                    continue
-                self.pool.apply_async(
-                    self._cache_morlist_raw_atomic,
-                    args=(i_key, 'compute_resource', compute_resource, tags_copy, regexes, include_only_marked)
-                )
+
+            for resource in obj.hostFolder.childEntity:
+                # Resource pool
+                if isinstance(resource, vim.ClusterComputeResource):
+                    self.pool.apply_async(
+                        self._cache_morlist_raw_atomic,
+                        args=(i_key, 'compute_resource', resource, tags_copy, regexes)
+                    )
+
+                # Folder pool
+                if isinstance(resource, vim.Folder):
+                    self.pool.apply_async(
+                        self._cache_morlist_raw_atomic,
+                        args=(i_key, 'folder', resource, tags_copy, regexes)
+                    )
 
         elif obj_type == 'compute_resource':
-            if obj.__class__ == vim.ClusterComputeResource:
-                cluster_tag = "vsphere_cluster:%s" % obj.name
-                tags_copy.append(cluster_tag)
+            cluster_tag = "vsphere_cluster:%s" % obj.name
```

Comment:
Are we sure that if `obj_type == 'compute_resource'` that implies `obj.__class__ == vim.ClusterComputeResource`.... not sure if pyvmomi might let some of the other objects fall into a compute_resource category (ie. datastores or whatever). We should make sure of this.